cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillController.java

```diff
@@ -140,10 +140,15 @@ public void complete() {
     } catch (Exception e) {
       // If there is exception, use the remote execution controller to try killing the remote process
       try {
-        LOG.debug("Force termination of remote process for program run {}", programRunId);
-        remoteProcessController.kill(RuntimeJobStatus.RUNNING);
+        RuntimeJobStatus currentStatus = remoteProcessController.getStatus();
```

Comment:
The Prior Logic : 

- While STATUS is RUNNING , keep checking every second. 
- If it exceeds for more than 5 seconds, then throw `IllegalStateException` 
- So the moment `getStatus` is called, the 5 second check is done without any gap and immediately goes to catch block for force termination. 

I agree between this few MS the dataproc job status could be `DONE`

But in the new extra check, the GAP for error still exists and this intermittent Wrong killing of pipeline would still happen. 

My point is there is not much time gap between the existing `getStatus == running` and remoteProcessController.kill() , and similar to extra check..